### PR TITLE
Implement structured output for `call` command

### PR DESCRIPTION
**Pull request by mkaput**
commit-id:62ce5726

---

**Stack**:
- #1146
- #1132
- #1137 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: protostar/commands/call_command.py [hunk 1]
```diff
@@ -58,41 +79,38 @@ def arguments(self):
         ]
 
     async def run(self, args: Any):
+        msg = OutputCommandUtil(args).get_messenger(self._log_color_provider)
```
Comment (Arcticae) [inline protostar/commands/call_command.py:82]: please give variables more meaningful names,  - `messenger()` would be nicer here

File: tests/e2e/test_gateway_commands.py [hunk 2]
```diff
@@ -36,7 +35,58 @@ def test_deploying_contract_with_constructor(
...
+            "--contract-address",
+            contract_address,
+            "--function",
+            "get_balance",
+            "--json",
+        ],
+        ignore_stderr=True,
+    )
+
+    assert result == """\
```
Comment (Arcticae) [inline tests/e2e/test_gateway_commands.py:87]: I'd parse it and compare dictionaries, since some whitespaces might change and the test is less volatile that way

**Review by Arcticae — APPROVED** [review]
LGTM

File: tests/e2e/test_gateway_commands.py (hunk 2 above)
Comment (mkaput) [reply tests/e2e/test_gateway_commands.py:87]: My idea is to also test whitespace. I want to ensure that output is not starting with whitespace, and the object is compressed as much as possible.

File: protostar/commands/call_command.py (hunk 1 above)
Comment (mkaput) [reply protostar/commands/call_command.py:82]: `write`? I would like to keep this a verb

File: protostar/commands/call_command.py [hunk 3]
```diff
@@ -1,24 +1,44 @@
...
 )
+from protostar.io import LogColorProvider, StructuredMessage
 from protostar.starknet_gateway import GatewayFacadeFactory
 
 
+@dataclass
+class SuccessfulCallMessage(StructuredMessage):
+    response: Any
+
+    def format_human(self, fmt: LogColorProvider) -> str:
```
Comment (MaksymilianDemitraszek) [inline protostar/commands/call_command.py:22]:
```suggestion
    def format_human(self, fmt: LogColorProvider) -> str:
```
`get_console_output` maybe?

File: protostar/commands/call_command.py [hunk 4]
```diff
@@ -1,24 +1,44 @@
...
+        return f"""\
+{fmt.colorize("GREEN", "Call successful.")}
+Response:
+{json.dumps(self.response._asdict(), indent=4)}\
+"""
+
+    def format_object(self) -> dict:
```
Comment (MaksymilianDemitraszek) [inline protostar/commands/call_command.py:29]:
```suggestion
    def as_dict(self) -> dict:
```
maybe?

File: website/docs/cli-reference.md [hunk 5]
```diff
@@ -38,6 +38,8 @@ The name of the function being called.
 The URL of a StarkNet gateway. It is required unless `--network` is provided.
 #### `--inputs FELT[]`
 Inputs to the function being called, represented by a list of space-delimited values.
+#### `--json`
+Enable JSON output format.
```
Comment (MaksymilianDemitraszek) [inline website/docs/cli-reference.md:42]:
```suggestion
Save output as json file.
```

File: protostar/commands/call_command.py (hunk 3 above)
Comment (mkaput) [reply protostar/commands/call_command.py:22]: get_console_output and as_dict are not consistently named.

I thought about `display_human` and `display_object` like Rust's `Display` trait, but I think format looks better.

File: protostar/commands/call_command.py (hunk 4 above)
Comment (mkaput) [reply protostar/commands/call_command.py:29]: https://github.com/software-mansion/protostar/pull/1137#discussion_r1018963715

File: website/docs/cli-reference.md (hunk 5 above)
Comment (mkaput) [reply website/docs/cli-reference.md:42]: We are not saving output anywhere here. I could say `Emit output as json file.` instead. BTW I copied help message from Cargo :P

Comment (mkaput) [reply website/docs/cli-reference.md:42]: Terraform has `If specified, machine readable output will be printed in JSON format.`. Maybe this one is better?

Comment (MaksymilianDemitraszek) [reply website/docs/cli-reference.md:42]: Yea i just understood you just print it and user can redirect it. I though protostar will save it to a file.

Comment (MaksymilianDemitraszek) [reply website/docs/cli-reference.md:42]: Maybe just `Outputs result in JSON format`

File: protostar/commands/call_command.py (hunk 4 above)
Comment (MaksymilianDemitraszek) [reply protostar/commands/call_command.py:29]: `format_dict` maybe? It returns dict not an object

File: website/docs/cli-reference.md (hunk 5 above)
Comment (mkaput) [reply website/docs/cli-reference.md:42]: done

File: protostar/commands/call_command.py (hunk 4 above)
Comment (mkaput) [reply protostar/commands/call_command.py]: done

File: protostar/commands/call_command.py (hunk 1 above)
Comment (mkaput) [reply protostar/commands/call_command.py:82]: done
